### ISSUE TITLE
Question: Disabling Eureka

### ISSUE THREAD

**Issue by balajivenki** [Netflix/ndbench#134]
@ipapapa  Looks like right after the comming #130 I see my local ndbench breaks due to Eureka dependency. While I resolved some of it using

eureka.validateInstanceId=false

with that the app runs. However I see these exceptions all the time

`13:12:18,230 ERROR DiscoveryClient:972 - DiscoveryClient_UNKNOWN/10.101.11.44 - was unable to refresh its cache! status = There is no known eureka server; cluster server list is empty
com.netflix.discovery.shared.transport.TransportException: There is no known eureka server; cluster server list is empty
	at com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient.execute(RetryableEurekaHttpClient.java:107)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.getApplications(EurekaHttpClientDecorator.java:134)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator$6.execute(EurekaHttpClientDecorator.java:137)
	at com.netflix.discovery.shared.transport.decorator.SessionedEurekaHttpClient.execute(SessionedEurekaHttpClient.java:77)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.getApplications(EurekaHttpClientDecorator.java:134)
	at com.netflix.discovery.DiscoveryClient.getAndStoreFullRegistry(DiscoveryClient.java:1051)
	at com.netflix.discovery.DiscoveryClient.fetchRegistry(DiscoveryClient.java:965)
	at com.netflix.discovery.DiscoveryClient.refreshRegistry(DiscoveryClient.java:1471)
	at com.netflix.discovery.DiscoveryClient$CacheRefreshThread.run(DiscoveryClient.java:1438)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
13:12:18,232 ERROR DiscoveryClient:860 - DiscoveryClient_UNKNOWN/10.101.11.44 - was unable to send heartbeat!
com.netflix.discovery.shared.transport.TransportException: There is no known eureka server; cluster server list is empty
	at com.netflix.discovery.shared.transport.decorator.RetryableEurekaHttpClient.execute(RetryableEurekaHttpClient.java:107)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.sendHeartBeat(EurekaHttpClientDecorator.java:89)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator$3.execute(EurekaHttpClientDecorator.java:92)
	at com.netflix.discovery.shared.transport.decorator.SessionedEurekaHttpClient.execute(SessionedEurekaHttpClient.java:77)
	at com.netflix.discovery.shared.transport.decorator.EurekaHttpClientDecorator.sendHeartBeat(EurekaHttpClientDecorator.java:89)
	at com.netflix.discovery.DiscoveryClient.renew(DiscoveryClient.java:846)
	at com.netflix.discovery.DiscoveryClient$HeartbeatThread.run(DiscoveryClient.java:1399)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)`

Is there any way to disable Eureka dependency or suppress these with an eureka config?

**Comment by ipapapa** [comment]
I will check but it does not seem it was #130 that caused the issue. That PR was only related white spaces and tabs.

**Comment by balajivenki** [comment]
Thanks @ipapapa 

I was referring to this commit

https://github.com/Netflix/ndbench/commit/7a8a443f864e694e7fa8aba90871404a780d0ef1

Looks like EurekaModule was added for EVCache plugin.

**Comment by ipapapa** [comment]
@balajivenki Can you point me where in that commit you see the change? I believe the `install(new EurekaModule());` was in another commit and we need to get it fixed for you.

**Comment by balajivenki** [comment]
@ipapapa I had the code running today-2days without any Eureka server in our eco system. All worked fine, yesterday I pulled the changes from master into my forked branch and right after that I saw the above errors. Upfront it did not work at all and after adding

eureka.validateInstanceId

it started working. But still I see that heartbeat is causing lots of exceptions when there is no Eureka in the eco system.
